rasa/core/policies/ted_policy.py

```diff
@@ -590,6 +633,56 @@ def train(
             shuffle=False,  # we use custom shuffle inside data generator
         )
 
+    def run_post_training_procedures(
+        self, model_data: RasaModelData, label_ids: np.ndarray
+    ) -> None:
+        """Run any tasks to be completed after the model has been trained.
+
+        Args:
+            model_data: Featurized data on which the model has been trained.
+            label_ids: Label ids corresponding to the featurized data.
+        """
+        # No post training procedure for `TEDPolicy`
+        return
+
+    def train(
+        self,
+        training_trackers: List[TrackerWithCachedStates],
+        domain: Domain,
+        interpreter: NaturalLanguageInterpreter,
+        **kwargs: Any,
+    ) -> None:
+        """Train the policy on given training trackers.
+
+        Args:
+            training_trackers: List of training trackers to be used
+                for training the model.
+            domain: Domain of the assistant.
+            interpreter: NLU Interpreter to be used for featurizing the states.
+            **kwargs: Any other argument.
+        """
+        if not training_trackers:
+            logger.error(
+                f"Can not train '{self.__class__.__name__}'. No data was provided. "
+                f"Skipping training of the policy."
+            )
+            return
+
+        model_data, label_ids = self._prepare_for_training(
+            training_trackers, domain, interpreter, **kwargs
+        )
+
+        if model_data.is_empty():
+            logger.error(
+                f"Can not train '{self.__class__.__name__}'. No data was provided. "
+                f"Skipping training of the policy."
+            )
+            return
+
+        self.run_training(model_data)
+
+        self.run_post_training_procedures(model_data, label_ids)
```

Comment:
Again, looks like this call is not needed.

Reply:
It's needed for `IntentTEDPolicy`. Any policy inheriting from `TEDPolicy` can have a post training procedure to run. `IntentTEDPolicy` does and that's why the call is needed.